### DISCUSSION TITLE
Page does not exist in "html" format Exception is thrown for templates that are configuard as headless.

### DISCUSSION THREAD

**Discussion by mario-fehr** [sulu/SuluHeadlessBundle#105] [Q&A] [answered]
| Q | A
| --- | ---
| Sulu Version |  2.4.0
| PHP Version | 8.0.2
| DB Version | MySQL 8
| Headless Bundle Version |  0.6.2

# Actual Behavior

If now twig file exist for the template the exception get thrown.
![image](https://user-images.githubusercontent.com/14167832/151529591-ae466e26-15e8-4567-ab82-d664c4e09f1b.png)
If the twig file exist for the template everthing is just fine.
![image](https://user-images.githubusercontent.com/14167832/151529794-296a11a6-3517-45e3-a3fd-74252df741f7.png)

# Expected Behavior

No need to create the twig template files if the template should be used as headless.

# Steps to Reproduce

Here is a [link](https://github.com/mfehr94/sulu-headless-demo) to the demo repo with the bug.

**Comment by alexander-schranz** [comment]
@mfehr94 the exception is only thrown when you are not requesting the page with `.json`. When using the headless bundle you need to make sure that you request the json using `/your-url.json`. Without it requires a `.html.twig` as you are then requesting a html response. So this behavoiur is expected.

If you use javascript server side (Next.js, Nuxt, ...) rendered implementation all request to `/your-url` (without .json) should be handled by that serve and so never hit sulu itself. This is something you need to make sure in your VHost or server routing. Else if sulu provide the html where a client side application is started to communicate over with the headless bundle create one `headless.html.twig` which is used by every template which just starts your client javascript.

**Comment by mario-fehr** [comment, marked as answer]
@alexander-schranz thank you for the explanation. My mistake was I did not change the `view` property in the template xml files. So there are two ways of doing this. Use `<view>base</view>` or ` <view>pages/headless</view>` if you want to use `templates/pages/headless.html.twig`
